### PR TITLE
improved error reporting on type mismatch of metrics

### PR DESCRIPTION
**Pull request by ohm314**

### CODE REVIEW

File: hpcbench/driver.py [hunk 1]
```diff
@@ -520,17 +520,23 @@ def _check_metrics(cls, schema, metrics):
...
             if not isinstance(value, metric.type):
-                message = "Unexpected type for metrics {}".format(name)
+                message = "Unexpected type for metrics {}:\n".format(name)
+                message += "expected {}, but got {}".format(metric.type,
+                                                            type(value))
                 raise Exception(message)
         elif isinstance(metric, list):
             if not isinstance(value, list):
                 message = "Unexpected type for metrics {}".format(name)
+                message += "expected {}, but got {}".format(type(list),
```
Comment (tristan0x) [inline hpcbench/driver.py]: The `type` builtin should not be used here. For instance:

```
>>> "{}".format(list)
"<type 'list'>"
>>> "{}".format(type(list))
"<type 'type'>"
>>> 
```

File: hpcbench/driver.py [hunk 2]
```diff
@@ -520,17 +520,23 @@ def _check_metrics(cls, schema, metrics):
...
+                                                            type(value))
                 raise Exception(message)
             for item in value:
                 cls._check_metric(schema, metric[0], name, item)
         elif isinstance(metric, dict):
             if not isinstance(value, dict):
                 message = "Unexpected type for metrics {}".format(name)
+                message += "expected {}, but got {}".format(type(dict),
```
Comment (tristan0x) [inline hpcbench/driver.py]: Same comment here, `type` should not be used.

Comment (ohm314) [reply hpcbench/driver.py]: yes, you're right of course.. when breaking the string into too, I did something stupid - I'll fix that.

**Comment by tristan0x** [comment]
LGTM!
Nice improvement @ohm314

**Pull request merged**